Gems/ROS2/Code/Source/FactorySimulation/ConveyorBeltComponentKinematic.cpp

```diff
@@ -0,0 +1,306 @@
+                        "physics engine.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &ConveyorBeltComponentKinematic::m_textureScale,
+                        "Texture scale",
+                        "Scale of the texture on conveyor belt.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &ConveyorBeltComponentKinematic::m_materialAsset,
+                        "Material",
+                        "Material asset of the conveyor belt")
+                    ->Attribute(AZ::Edit::Attributes::DefaultAsset, &GetDefaultPhysicsMaterialAssetId)
+                    ->Attribute(AZ_CRC_CE("EditButton"), "")
+                    ->Attribute(AZ_CRC_CE("EditDescription"), "Open in Asset Editor")
+                    ->Attribute(AZ_CRC_CE("DisableEditButtonWhenNoAssetSelected"), true);
+            }
+        }
+    }
+
+    void ConveyorBeltComponentKinematic::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("SplineService"));
+    }
+
+    void ConveyorBeltComponentKinematic::Activate()
+    {
+        m_initialized = false;
+        AZ::TickBus::Handler::BusConnect();
+    }
+
+    void ConveyorBeltComponentKinematic::Deactivate()
+    {
+        m_sceneFinishSimHandler.Disconnect();
+        AZ::TickBus::Handler::BusDisconnect();
+    }
+
+    AZ::Vector3 ConveyorBeltComponentKinematic::GetLocationOfSegment(
+        AzPhysics::SceneHandle sceneHandle, AzPhysics::SimulatedBodyHandle handle)
+    {
+        AzPhysics::SceneInterface* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        AZ_Assert(sceneInterface, "No scene interface");
+        auto* body = azdynamic_cast<AzPhysics::RigidBody*>(sceneInterface->GetSimulatedBodyFromHandle(sceneHandle, handle));
+        AZ_Assert(body, "No valid body found");
+        if (body)
+        {
+            AZ::Vector3 beginOfSegments = body->GetPosition();
+            return beginOfSegments;
+        }
+        return AZ::Vector3::CreateZero();
+    }
+
+    AZStd::pair<float, AzPhysics::SimulatedBodyHandle> ConveyorBeltComponentKinematic::CreateSegment(
+        AZ::ConstSplinePtr splinePtr,
+        AzPhysics::SystemInterface* physicsSystem,
+        AzPhysics::SceneHandle sceneHandle,
+        float normalizedLocation)
+    {
+        auto colliderConfiguration = AZStd::make_shared<Physics::ColliderConfiguration>();
+        colliderConfiguration->m_materialSlots.SetMaterialAsset(0, m_materialAsset);
+        auto shapeConfiguration =
+            AZStd::make_shared<Physics::BoxShapeConfiguration>(AZ::Vector3(m_segmentLength, m_beltWidth, m_segmentWidth));
+        const auto transform = GetTransformFromSpline(m_splineConsPtr, normalizedLocation);
+        AzPhysics::RigidBodyConfiguration conveyorSegmentRigidBodyConfig;
```

Comment:
Populating fields of a single segment is a good candidate for a subroutine to help with readibility

Reply:
True, will add.